polaris-core/src/main/java/org/apache/polaris/core/connection/IcebergCatalogPropertiesProvider.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.connection;
+
+import jakarta.annotation.Nonnull;
+import java.util.Map;
+import org.apache.polaris.core.secrets.UserSecretsManager;
+
+public interface IcebergCatalogPropertiesProvider {
+  @Nonnull
+  Map<String, String> asIcebergCatalogProperties(UserSecretsManager secretsManager);
```

Comment:
I wonder why this is not just part of the functionality of `UserSecretsManager`... can you add more color here?

Reply:
maybe in the javadoc :)